bootstrap.sh

```diff
@@ -1,39 +1,505 @@
+    echoInfo "Terraform is distributed as a single binary. Install Terraform by unzipping it and moving it to a directory included in your system's PATH ."
+}
+
+install_tf() {
+    echoInfo "====> Installing terraform..."
+    if [[ "$1" == "windows" ]]; then
+        help_install_tf
+        return 1
+    fi
+    if [[ -z "$(command -v unzip)" ]]; then
+        {
+            sudo apt-get update -qy && sudo apt-get install -qy unzip > /dev/null
+        } || {
+            sudo apt update -qy && sudo apt install -qy unzip > /dev/null
+        } || {
+            sudo yum update -qy && sudo yum install -qy unzip > /dev/null
+        } || {
+            brew tap homebrew/dupes && brew install unzip > /dev/null
+        } || {
+            echoError "Could not install unzip"
+            help_install_tf
+            return 1
+        }
+        
+    fi
+    curl -fSL -o terraform.zip https://releases.hashicorp.com/terraform/"$TERRAFORM_VERSION"/terraform_"$TERRAFORM_VERSION"_"$1"_amd64.zip
+    sudo mkdir -p "$LIB_LOCATION"/
+    sudo unzip -q terraform.zip -d "$LIB_LOCATION"/terraform
+    rm -f terraform.zip
+    sudo ln -Ffs "$LIB_LOCATION"/terraform/terraform /usr/local/bin/terraform
+    if [[ -z $(command -v terraform) ]]; then
+        help_install_tf
+        return 1
+    else
+        echoSuccess "terraform installed"
+        terraform --version
+        echo ""
+        return 0
+    fi
+}
+
+check_tf() {
+    { # Try
+        if [[ -z $(command -v terraform) ]]; then
+            if [[ "$OSTYPE" == "linux-gnu" ]]; then
+                install_tf "linux"
+            elif [[ "$OSTYPE" == "darwin"* ]]; then
+                # Mac OSX
+                install_tf "darwin"
+            elif [[ "$OSTYPE" == "cygwin" ]]; then
+                # POSIX compatibility layer and Linux environment emulation for Windows
+                install_tf "windows"
+            elif [[ "$OSTYPE" == "msys" ]]; then
+                # Lightweight shell and GNU utilities compiled for Windows (part of MinGW)
+                install_tf "windows"
+            elif [[ "$OSTYPE" == "win32" ]]; then
+                # I'm not sure this can happen.
+                install_tf "windows"
+            elif [[ "$OSTYPE" == "freebsd"* ]]; then
+                install_tf "freebsd"
+            else
+                help_install_tf
+                return 1
+            fi
+        else
+            echoSuccess "terraform found in path!"
+            terraform --version    
+            echo ""
+            return 0            
+        fi
+        return $?
+    } || { # Catch
+        help_install_tf
+        return 1
+    }
+}
+
+#
+# Check between apt or yum
+#
+install_iofogctl_linux() {
+    if [[ -x "$(command -v apt-get)" ]]; then
+        curl -s https://packagecloud.io/install/repositories/iofog/iofogctl/script.deb.sh | sudo bash
+        sudo apt-get install iofogctl -y
+    elif [[ -x "$(command -v apt)" ]]; then
```

Comment:
this fork may be redundant, not sure, but is there is system that has apt but doesn't have apt-get?

Reply:
I have no idea, so I put it there in case of x)